src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,104 @@
+package duke;
+
+import duke.command.Command;
+
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * Utility class for formatting and printing output from Duke.
+ *
+ * @see Duke
+ */
+public class Ui {
+
+    private String[] helpMsg = {};
+
+    /**
+     * Defines the Duke commands from which the help message is generated.
+     *
+     * @param commands The commands recognized by Duke.
+     */
+    public void setCommands(Command[] commands) {
```

Comment:
Perhaps it would be more intuitive to rename the method as `setHelpMsg` instead? As setter methods should have names that describe what they are setting